### PR TITLE
Spine social button

### PR DESCRIPTION
**Pull request by galanovnick**
This PR introduce the `spine-social-button` element.

This element represents a raised `paper-button` equipped with an icon. An icon is separated from the button content (button text) by the dark separator.

There is the default styling for certain set of providers.
For now, following providers are supported: Google, Github, Facebook, Twitter.
Styles and icon are fully customizable.

@dpikhulya PTAL

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+The following custom properties and mixins are available for styling:
+
+Custom property | Description | Default
+----------------|-------------|----------
+`--spine-social-button-background` | Background color of the button | specified according to `provider`
+`--spine-social-button` | Mixin applied to the <paper-button> | `{}`
+`--spine-social-button-disabled` | Mixin passed to the `--paper-button-disabled` mixin| `{}`
+`--spine-social-button-focus` | Mixin passed to the `--paper-button-raised-keyboard-focus` mixin | `{}`
```
Comment (dpikhulya) [inline README.md]: Why don't we name it --spine-social-button-**keyboard**-focus to make the naming uniform with that from `paper-button` (especially if we're just delegating this to an analogous `paper-button`'s mixin)?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  <spine-social-button icon="my-icons:twitter">Share</spine-social-button>
+```
+
+The following custom properties and mixins are available for styling:
+
+Custom property | Description | Default
+----------------|-------------|----------
+`--spine-social-button-background` | Background color of the button | specified according to `provider`
+`--spine-social-button` | Mixin applied to the <paper-button> | `{}`
+`--spine-social-button-disabled` | Mixin passed to the `--paper-button-disabled` mixin| `{}`
```
Comment (dpikhulya) [inline README.md]: Missing space before he ending `|` symbol

File: spine-social-button.html [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  <spine-social-button icon="my-icons:twitter">Share</spine-social-button>
+```
+
+The following custom properties and mixins are available for styling:
+
+Custom property | Description | Default
+----------------|-------------|----------
+`--spine-social-button-background` | Background color of the button | specified according to `provider`
+`--spine-social-button` | Mixin applied to the <paper-button> | `{}`
+`--spine-social-button-disabled` | Mixin passed to the `--paper-button-disabled` mixin| `{}`
```
Comment (dpikhulya) [inline spine-social-button.html]: Make sure to apply README.md fixes here (whenever applicable).

File: spine-social-button.html [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+      iron-icon {
+        padding-right: 8px;
+      }
+
+      #divider {
+        display: inline-block;
+        width: 1px;
+        height: 100%;
+        background: rgba(0, 0, 0, .12);
```
Comment (dpikhulya) [inline spine-social-button.html]: Is --dark-divider-opacity applicable here? We would get rid of one more "magical" constant this way.

**Review by dpikhulya — COMMENTED** [review]
Please see a couple of comments.

**Comment by galanovnick** [comment]
@dpikhulya PTAL

**Review by dpikhulya — APPROVED** [review]
@galanovnick LGTM

**Pull request merged**